File: app/src/main/java/com/ianarbuckle/dublinbushelper/firebase/authentication/FirebaseAuthHelper.java
package com.ianarbuckle.dublinbushelper.firebase.authentication;

import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
import com.ianarbuckle.dublinbushelper.firebase.RequestListener;

/**
 * Created by Ian Arbuckle on 04/03/2017.
 *
 */

public interface FirebaseAuthHelper {
  void googleLogin(GoogleSignInAccount account, RequestListener listener);
  void registerUser(String email, String password, final RequestListener listener);
  void signInUser(String email, String password, final RequestListener listener);
  void logOutUser();
  boolean isUserSignedIn();
  String getUserDisplayName();
  String getUserPhoto();
  String getUserEmail();

}

File: app/src/main/java/com/ianarbuckle/dublinbushelper/firebase/authentication/FirebaseAuthHelper.kt
package com.ianarbuckle.dublinbushelper.firebase.authentication

import com.google.android.gms.auth.api.signin.GoogleSignInAccount
import com.ianarbuckle.dublinbushelper.firebase.RequestListener

/**
 * Created by Ian Arbuckle on 04/03/2017.

 */

interface FirebaseAuthHelper {
    fun googleLogin(account: GoogleSignInAccount, listener: RequestListener)
    fun registerUser(email: String, password: String, listener: RequestListener)
    fun signInUser(email: String, password: String, listener: RequestListener)
    fun logOutUser()
    val isUserSignedIn: Boolean
    val userDisplayName: String?
    val userPhoto: String?
    val userEmail: String?
}
